test/e2e/tests/test_endpoint.py

```diff
@@ -0,0 +1,167 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# 	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the EventBridge Endpoint API.
+"""
+
+import pytest
+import time
+import logging
+
+from acktest.resources import random_suffix_name
+from acktest.k8s import resource as k8s
+from acktest.k8s import condition as condition
+from acktest.aws.identity import get_region
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_eventbridge_resource
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.tests.helper import EventBridgeValidator
+from e2e.bootstrap_resources import get_bootstrap_resources
+
+RESOURCE_PLURAL = "endpoints"
+
+CREATE_WAIT_AFTER_SECONDS = 10
+UPDATE_WAIT_AFTER_SECONDS = 10
+DELETE_WAIT_AFTER_SECONDS = 10
+
+def create_event_bus(aws_name: str, region: str):
+    resource_name = random_suffix_name("ack-test-bus", 24)
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["BUS_NAME"] = aws_name
+
+    # Load EventBus CR
+    resource_data = load_eventbridge_resource(
+        "eventbus",
+        additional_replacements=replacements,
+    )
+    logging.debug(resource_data)
+
+    # Create k8s resource
+    resource_data["metadata"] = {
+        "name": resource_name,
+        "annotations": {"services.k8s.aws/region": region},
+    }
+
+    ref = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, "eventbuses",
+        resource_name, namespace="default",
+    )
+    k8s.create_custom_resource(ref, resource_data)
+    cr = k8s.wait_resource_consumed_by_controller(ref)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(ref)
+
+    time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+    cr = k8s.wait_resource_consumed_by_controller(ref)
+
+    return ref, cr
+
+@pytest.fixture(scope="module")
+def bus_name():
+    return random_suffix_name("bus", 24)
```

Comment:
Thank you for the tips, senpai @jaypipes !

Reply:
Those review comments, hilarious xD

Reply:
> Thank you for the tips, senpai @jaypipes !

Very welcome, @A-Hilaly :)